### ISSUE TITLE
Floating base parts

### ISSUE THREAD

**Issue by codexGW** [A3Wasteland/ArmA3_Wasteland.Altis#448]
If there's a way someone can fix the immersion breaking base parts floating everywhere, or add a option to delete them maybe through the admin camera, it would be appreciated.

**Comment by ghost** [comment]
Admin camera lol, the stock admin tools are poor at best, 

You could either write a delete cursor target option or take the delete object script from my admin tools chap

**Comment by codexGW** [comment]
The admin camera is already part of the admin tools. I helped micovery test port it over. The current ones are pretty good if you add a couple easy things to them. Most just copy what louDnl did. We set our deletevehicle to a limit so that would cause kicks a lot of the time. Trust me I spent a couple days trying to add some kind of delete cursor target tool. I'm not really good a coding if I don't have an example to just copy. The idea of just adding it to the admin cam is something I got from infiSTAR.

**Comment by ghost** [comment]
Well I've spent the last week or so tinkering with mine, 

New admin GUI and playerSelect GUI
Map esp
Esp
Ai esp
A proper spectate tool
Kick
New TP 2 me to them as the original was broke following the last update.
Delete cursor target
Repair cursor target

Lots of other tools to,

**Comment by ghost** [comment]
But now you mention it a feecam option would also be a good idea.

**Comment by codexGW** [comment]
Yeah you told me to join your server to see them. That's not the easiest way to see anything you do. I have to go through a bunch of stuff just to see what you did and use some kind of diff program like beyond compare. This is what I did to the admin panel basically coping what loudnl and others have done.
https://github.com/A3Armory/A3A_A3Wasteland.Altis/commit/2032cf0de05e63d176fb0260fd393cef67257542
Real easy to share that.

**Comment by ghost** [comment]
oh well I can't help you in that that respect as I'm laid in bed :-)

I'll upload my mission to GitHub tomorrow

**Comment by codexGW** [comment]
Don't worry about that. It helps when reporting issues or talking about things you have done. I don't know how you keep track of it really. It's for you though and it's more about the commits. Loudnl and TOP Arma really did a good job of this and they are still helping the A3W community (at least me) today with their nicely done commits.

**Comment by ghost** [comment]
I'm the prodigal Frankenstein coder lol, It's quite easy in my own way of doing things tbh. 

Torn and Maca don't do things for me except point me in the right direction as 'that's the only way I'll learn' grrr :-)

**Comment by hobart666** [comment]
Hey Matt I've just recently noticed the old script that tptome and such uses for the player list array is broken as well. And have been beating my head on the desk trying to finger out that one. Is there any chance I could have a look at What you have going on for that? I dig what you say about Torn and MAca, LouD did the same to me but now he's not interested at all sadly poor guy got burnt out from arma I think. I need those pointing again... I miss them. :)

**Comment by codexGW** [comment]
That whole script is a bit wonky. Try the commit here if you want it in the player management and logged for misuse. https://github.com/A3Armory/A3A_A3Wasteland.Altis/commit/2032cf0de05e63d176fb0260fd393cef67257542

**Comment by hobart666** [comment]
That is much tidier than that seriously old & as you say wonky, dude that's very wonky to say the least it's old fusionsmenu script from a2 lol Thank you sir.

**Comment by WhatsUpDoc127** [comment]
Gigatek in the wasteland confit you need to disable spawn base parts in town. Then there is Matt's Object Deleter that can be added to the admin panel so any admin can delete any object on the map that players leave laying around like walls. I'm on the wall but will post how to enable it after I get home

**Comment by WhatsUpDoc127** [comment]
Above I meant to say WASTELAND SERVER CONFIG. Disable spawning base parts in towns!!!

**Comment by WhatsUpDoc127** [comment]
Here is how to implement Matt's Object Deleter which will allow you to delete objects and vehicles:
Go to client\systems\adminPanel\loadServerAdministratorMenu.sqf

`#define adminMenu_option 50001
disableSerialization;

private ["_start","_panelOptions","_displayAdmin","_adminSelect"];
_uid = getPlayerUID player;
if ([_uid, 3] call isAdmin) then {
    _start = createDialog "AdminMenu";

```
_displayAdmin = uiNamespace getVariable "AdminMenu";
_adminSelect = _displayAdmin displayCtrl adminMenu_option;

_panelOptions = ["Player Management",
                "Vehicle Management",
                "Map Markers Log",
                "Group Leader Markers",
                "Teleport",
                                "Money",
                                "Debug Menu",
                "Object Search",
                               "Toggle God-mode",
                **"Matt76 Object Deleter",**
                "Teleport - player to me",
                "Teleport - me to player",
                "Toggle Invisible-mode"`
```

See where I have my located starting from Player Management at line 0 down its at line 9.

Then you have to open your optionsSelect.sqf in the same directory and since Matt76 Ojbect Deleter is on line 9 I you have to put it in the same case # which equals the same line number for example:

`case 0: //Player Menu
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\playerMenu.sqf";
                };
                case 1: //Full Vehicle Management
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\vehicleManagement.sqf";
                };
                case 2: //Markers log
                {
                    closeDialog 0;
                    createDialog "MarkerLog";
                };
                case 3: //Tags
                {
                    execVM "client\systems\adminPanel\playerTags.sqf";
                };
                case 4: //Teleport
                {
                    closeDialog 0;
                    ["A3W_teleport", "onMapSingleClick",
                    {
                        vehicle player setPos _pos;
                        //if (!isNil "notifyAdminMenu") then { ["teleport", _pos] spawn notifyAdminMenu };
                        ["A3W_teleport", "onMapSingleClick"] call BIS_fnc_removeStackedEventHandler;
                        true
                    }] call BIS_fnc_addStackedEventHandler;
                    hint "Click on map to teleport";
                };
                case 5: //Money
                {
                    _money = 20000;
                    player setVariable ["cmoney", (player getVariable ["cmoney",0]) + _money, true];
                    //if (!isNil "notifyAdminMenu") then { ["money", _money] call notifyAdminMenu };
                };
                case 6: //Debug Menu
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\loadDebugMenu.sqf";
                };
                case 7: //Object search menu
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\loadObjectSearch.sqf";
                };
                case 8: // toggle God mode
                {
                    execVM "client\systems\adminPanel\toggleGodMode.sqf";
                };
                case 9: // delete base parts or vehicles
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\deleteCursor.sqf";
                };
                case 10: // Teleport - player to me
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\tptome.sqf";
                    hint "Select player from drop-down list on left to TELEPORT to YOUR position!";
                };
                case 11: // Teleport - me to player
                {
                    closeDialog 0;
                    execVM "client\systems\adminPanel\tpmeto.sqf";
                    hint "Select player from drop-down list on left to TELEPORT to THEIR position!";
                };
                case 12: //Invisible Mode 
                {
                    execVM "client\systems\adminPanel\toggleInvisMode.sqf";
                };`

Keep in mind if your going to use my full admin menu the Teleport player to me and Teleport me to player DOES NOT WORK at this time thanks to Bohemia after the 1.64 update. I wish Creampie, LouD or whoever created these two scripts would look through them and see if they could fix them to make them work but as of right now they wont work. They will cause the game to shutdown and you get an Access Error. I may sift through to see what I can do but I am working on something else at the moment. Anyway just use the Matt76 Deleter to deleter to delete whatever you need. The deleter says in your menu for 30 seconds and it will disappear once it does you have to bring back up the admin menu to enable it again. If you want extend how long it stays enabled just edit the deleteCursor.sqf....

Hope this helps.

You can send me a PM on wasteland forums and I will send you the files or proper code as GITHUB screws up the way the code is suppose look on here for some reason!!!!! Sorry

**Comment by codexGW** [comment]
Thanks for the details man. I'll check out the deleteCursor.sqf. For a working tp to me and them check this out. Its done through the player management and logged completely.
https://github.com/A3Armory/A3A_A3Wasteland.Altis/commit/2032cf0de05e63d176fb0260fd393cef67257542

**Comment by WhatsUpDoc127** [comment]
If you ressearch LouDs Altis 1.2 wasteland the Matt76 Object Deleter is in there if I am not mistaken... Thanks for the heads up on hte tp to me..

**Comment by hobart666** [comment]
`deleteVehicle cursorTarget;`

and name it deleteCursor.sqf or w/e you wish.

**Comment by ghost** [comment]
That's my old old menu whatsupdoc, I've got all new tools now ;-)

**Comment by WhatsUpDoc127** [comment]
ccgmatt76 it still works with 1.3b!!!! and we still use it. Can you point me to your new tools?

**Comment by WhatsUpDoc127** [comment]
Are your new news in the Tanoa or New Stratis on your git?

**Comment by WhatsUpDoc127** [comment]
NVM just found them...thanks...

**Comment by ghost** [comment]
https://github.com/ccgmatt76/Admin-Tools

I will pop this on the A3 forums later and let people pick what they want, a little bit of editing is required for other communities to full use these

**Comment by WhatsUpDoc127** [comment]
matt do you have an email address because I need some help on a script that I cannot get to work. Or can I PM you on Wasteland Forums?

**Comment by ghost** [comment]
Add me on steam chap Matt76 I may well be online later but I get messages on my mobile also
